src/kernels/MIOpenCol2Im2d.cl

```diff
@@ -33,51 +33,61 @@
 #define ACCUMULATOR_NEEDS_CONVERSION 0
 #endif
 
-__kernel void Col2Im2d(global _FLOAT* col,
-                       const int col_h,
-                       const int col_w,
-                       const int wei_h,
-                       const int wei_w,
-                       const int pad_h,
-                       const int pad_w,
-                       const int stride_h,
-                       const int stride_w,
-                       const int dilation_h,
-                       const int dilation_w,
-                       const int height,
-                       const int width,
-                       global _FLOAT* im,
-                       const int im_offset)
+#ifndef MIOPEN_USE_64BIT_INDEX
+#error "MIOPEN_USE_64BIT_INDEX must be defined"
+#endif
+
+#if MIOPEN_USE_64BIT_INDEX
+typedef ulong index_t;
+#else
+typedef uint index_t;
+#endif
+
+__kernel void Col2Im2dU(global _FLOAT* col,
+                        const uint col_h,
+                        const uint col_w,
+                        const uint wei_h,
+                        const uint wei_w,
+                        const uint pad_h,
+                        const uint pad_w,
+                        const uint stride_h,
+                        const uint stride_w,
+                        const uint dilation_h,
+                        const uint dilation_w,
+                        const uint height,
+                        const uint width,
+                        global _FLOAT* im,
+                        const uint im_offset)
 {
     global _FLOAT* im_off = im + im_offset;
-    int gid               = (int)get_global_id(0);
+    uint gid              = (uint)get_global_id(0);
```

Comment:
All offset math switched to unsigned domain, which doubles the upper limit for free.